flink-core/src/main/java/org/apache/flink/core/plugin/PluginConfig.java

```diff
@@ -55,23 +55,22 @@ private PluginConfig(Optional<Path> pluginsPath, String[] alwaysParentFirstPatte
 
 	public static PluginConfig fromConfiguration(Configuration configuration) {
 		return new PluginConfig(
-			getPluginsDirPath(configuration),
+			getPluginsDir().map(File::toPath),
 			CoreOptions.getParentFirstLoaderPatterns(configuration));
 	}
 
-	private static Optional<Path> getPluginsDirPath(Configuration configuration) {
-		String pluginsDir = configuration.getString(ConfigConstants.ENV_FLINK_PLUGINS_DIR, null);
-		if (pluginsDir == null) {
-			LOG.info("Environment variable [{}] is not set", ConfigConstants.ENV_FLINK_PLUGINS_DIR);
-			return Optional.empty();
-		}
+	public static Optional<File> getPluginsDir() {
```

Comment:
nit: `Path` is considered the more modern API.

https://stackoverflow.com/questions/6903335/java-7-path-vs-file

The code below can be rewritten to:

```
final Path pluginsDirFile = Paths.get(pluginsDir);
if (!Files.isDirectory(pluginsDirFile)) {
```

Other places would need to be rewritten, however. Therefore, I am not insisting on a change.

Reply:
I deliberately used `File` in order to express that the directory needs to be local to the process. Happy to change it at a later point when we decide that the plugins directory can reside on any filesystem.

Reply:
Ok I am fine with keeping the code. However, I don't think that `java.nio.file.Path` conveys that the file can be potentially on a remote filesystem. On my platform there are only local `java.nio.file.FileSystem` implementations.

Reply:
True, but this must not hold for every other setup.